AsyncDisplayKit/ASButtonNode.mm

```diff
@@ -491,10 +491,17 @@ - (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
     spec = [ASInsetLayoutSpec insetLayoutSpecWithInsets:contentEdgeInsets child:spec];
   }
   
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
   if (CGSizeEqualToSize(self.preferredFrameSize, CGSizeZero) == NO) {
-    stack.sizeRange = ASRelativeSizeRangeMakeWithExactCGSize(self.preferredFrameSize);
+#if DEBUG
```

Comment:
I was under the impression that this new API is a replacement for PFS and we're simply keeping around PFS as a deprecated property. Is this not correct?
  
  

  
>   
> On Sep 5, 2016 at 01:43,  <Michael Schneider (mailto:notifications@github.com)>  wrote:
>   
>   
>   
>
> In   AsyncDisplayKit/ASButtonNode.mm (https://github.com/facebook/AsyncDisplayKit/pull/2110#discussion_r77487854):
>
>   
>   >  if (CGSizeEqualToSize(self.preferredFrameSize, CGSizeZero) == NO) {  >  - stack.sizeRange = ASRelativeSizeRangeMakeWithExactCGSize(self.preferredFrameSize);  >  +#if DEBUG  
>   
>
> So you are on the side to remove all usages in e.g.  ASTextNode,  ASButtonNode,  ASImageNode  ... the "problem" with that is just that this would break a lot of code and  preferredFrameSize  would basically be useless and we could remove it in the first place. I think it would make even more sense to remove it at all if we decide to not gracefully remove it in ASDK code as than developer would get an error and have to fix the places instead of just a warning or nothing if they don't have Xcode warnings enabled for deprecation.
>
>   
>
> —
>  You are receiving this because you were mentioned.
>  Reply to this email directly,   view it on GitHub (https://github.com/facebook/AsyncDisplayKit/pull/2110/files/ea5924cf576ad1f72241bc2751e5d1a0300ff8cc#r77487854), or   mute the thread (https://github.com/notifications/unsubscribe-auth/AAAhLSSMnAfLNc_95uxX93lr0aNEAEvIks5qm9Y6gaJpZM4Jo8Oy).
>
>   
>   
>     
>   
>   
>